nif_points_concat_node:
  ros__parameters:
    # point concat filter
    use_new_luminar_driver : false
    input_topics : "[/luminar_front_points, /luminar_left_points, /luminar_right_points]"
    output_topic : "/merged/lidar"

nif_ego_shape_filter_node:
  ros__parameters:
    # ego-shape filter
    left_lower_distance : 1.0
    right_lower_distance : 1.0
    rear_lower_distance : 3.2
    front_lower_distance : 2.5
    left_upper_distance : 30.0
    right_upper_distance : 30.0
    rear_upper_distance : 20.0
    front_upper_distance : 80.0
    height_lower_distance : -0.5
    height_upper_distance : 1.5
    resolution : 0.5 #0.25
    count_threshold : 0.1 #origin 3.


nif_wall_detection_node:
  ros__parameters:
    rear_upper_distance : 20.0
    front_upper_distance : 80.0
    normal_angle_thres : 70. #50.
    ransac_pts_thresh : 50  # 200
    ransac_distance_thres : 0.2 #0.2

    x_roi : 0.0
    distance_extract_thres : 0.5
    distance_low_fass_filter : 0.0
    target_space_to_wall : 3.0 #should be positive value

    # control params
    min_lookahead: 4.0
    max_lookahead: 29.0
    lookahead_speed_ratio: 0.75
    proportional_gain: .2
    max_steer_angle: 25.0
    vehicle:
      wheelbase: 2.9718
      cg_to_front: 1.723644
      cg_to_rear: 1.248156
      tire_stiffness_front: 100000.0
      tire_stiffness_rear: 100000.0
      mass: 720.0
      inertia: 1000.0

